app/assets/v2/js/cart.js

```diff
@@ -0,0 +1,788 @@
+        delete donation.name;
+        delete donation.grant;
+        delete donation.comment;
+        delete donation.tokenApprovalTxHash;
+        return donation;
+      });
+
+      // Remove donations of zero value
+      const donationInputsFiltered = donationInputs.filter(donation => {
+        return Number(donation.amount) !== 0;
+      });
+
+      indicateMetamaskPopup();
+      bulkTransaction.methods
+        .donate(donationInputsFiltered)
+        .send({ from: userAddress, gas: this.donationInputsGasLimit, value: this.donationInputsEthAmount })
+        .on('transactionHash', (txHash) => {
+          console.log('Donation transaction hash: ', txHash);
+          indicateMetamaskPopup(true);
+          this.postToDatabase(txHash, userAddress);
+          // Clear cart, redirect back to grants page, and show success alert
+          localStorage.setItem('contributions_were_successful', 'true');
+          localStorage.setItem('contributions_count', String(this.grantData.length));
+          this.clearCart();
+          var network = document.web3network;
+
+          if (network === 'rinkeby') {
+            window.location.href = `${window.location.origin}/grants/?network=rinkeby&category=`;
+          } else {
+            window.location.href = `${window.location.origin}/grants`;
+          }
+        })
+        .on('error', (error, receipt) => {
+          // If the transaction was rejected by the network with a receipt, the second parameter will be the receipt.
+          this.handleError(error);
+        });
+    },
+
+    postToDatabase(txHash, userAddress) {
+      // this.donationInputs is the array used for bulk donations
+      // We loop through each donation and POST the required data
+      const donations = this.donationInputs;
+      const csrfmiddlewaretoken = document.querySelector('[name=csrfmiddlewaretoken]').value;
+
+      for (let i = 0; i < donations.length; i += 1) {
+        // Get URL to POST to
+        const donation = donations[i];
+        const grantId = donation.grant.grant_id;
+        const grantSlug = donation.grant.grant_slug;
+        const url = `/grants/${grantId}/${grantSlug}/fund`;
+
+
+        // Get token information
+        const tokenName = donation.grant.grant_donation_currency;
+        const tokenDetails = this.getTokenByName(tokenName);
+
+        // Gitcoin uses the zero address to represent ETH, but the contract does not. Therefore we
+        // get the value of denomination and token_address using the below logic instead of
+        // using tokenDetails.addr
+        const isEth = tokenName === 'ETH';
+        const tokenAddress = isEth ? '0x0000000000000000000000000000000000000000' : tokenDetails.addr;
+
+        // Replace undefined comments with empty strings
+        const comment = donation.comment === undefined ? '' : donation.comment;
+
+        // For automatic contributions to Gitcoin, set 'gitcoin-grant-input-amount' to 100.
```

Comment:
The assumption here is that if someone makes a grant contribution they won't set the Gitcoin optional donation to 100%? Just checking my understanding here